### PR TITLE
Backport assisted linter fix from master to ocm-2.3 branch

### PR DESCRIPTION
**Pull request by YuviGold**
Master PR - https://github.com/openshift/release/pull/18965

/cc @osherdp  @lranjbar

### CODE REVIEW

File: ci-operator/config/openshift/assisted-installer/openshift-assisted-installer-ocm-2.3.yaml [hunk 1]
```diff
@@ -54,40 +53,25 @@ resources:
...
   commands: make REPORTS=${ARTIFACT_DIR} unit-test
   container:
-    from: bin
-- as: e2e-metal-assisted
-  steps:
-    cluster_profile: packet
-    dependencies:
-      ASSISTED_CONTROLLER_IMAGE: pipeline:assisted-installer-controller
-      ASSISTED_INSTALLER_IMAGE: pipeline:assisted-installer
-    workflow: baremetalds-assisted-install
```
Comment (osherdp) [inline ci-operator/config/openshift/assisted-installer/openshift-assisted-installer-ocm-2.3.yaml:76]: I'm not sure about having no e2e flow at all
Even though it's REST API, don't we want to make sure installer and controller are runnable in a full flow?

File: ci-operator/config/openshift/assisted-installer/openshift-assisted-installer-ocm-2.3.yaml [hunk 2]
```diff
@@ -54,40 +53,25 @@ resources:
 tag_specification:
   name: "4.8"
   namespace: ocp
+test_binary_build_commands: |
+  go mod vendor
+  mkdir /.cache && chmod 775 -R /.cache && chmod 775 -R $GOPATH
```
Comment (osherdp) [inline ci-operator/config/openshift/assisted-installer/openshift-assisted-installer-ocm-2.3.yaml:58]:
```suggestion
  mkdir /.cache && chmod 775 -R /.cache $GOPATH
```

File: ci-operator/config/openshift/assisted-installer-agent/openshift-assisted-installer-agent-ocm-2.3.yaml [hunk 3]
```diff
@@ -51,22 +50,31 @@ resources:
...
-    from: bin
+    from: test-bin
 - as: e2e-metal-assisted
   steps:
     cluster_profile: packet
     dependencies:
       ASSISTED_AGENT_IMAGE: pipeline:assisted-installer-agent
+    env:
+      ASSISTED_CONFIG: |
+        OPENSHIFT_VERSION=""
```
Comment (osherdp) [inline ci-operator/config/openshift/assisted-installer-agent/openshift-assisted-installer-agent-ocm-2.3.yaml:75]: we can run here just 4.8

**Review by osherdp — CHANGES_REQUESTED** [review]

File: ci-operator/config/openshift/assisted-installer/openshift-assisted-installer-ocm-2.3.yaml (hunk 1 above)
Comment (YuviGold) [reply ci-operator/config/openshift/assisted-installer/openshift-assisted-installer-ocm-2.3.yaml:76]: You are right. Bad copy-pasta

**Review by osherdp — APPROVED** [review]

**Comment by YuviGold** [comment]
Linter is expected to fail.
e2e jobs are flaky for an unknown reason. But sometimes it pass. I don't think it should be blocking.
/unhold

**Pull request merged**